File: app/src/main/java/br/com/onibusapp/onibusapp/data/SQLiteHelper.java
package br.com.onibusapp.onibusapp.data;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

/**
 * Created by diego on 18/07/2018.
 */

public class SQLiteHelper  extends SQLiteOpenHelper {

    private String[] scriptSQLCreate;
    private String scriptSQLDelete;

    public SQLiteHelper(Context context, String name, int version, String[] scriptSQLCreate,
                        String scriptSQLDelete) {
        super(context, name, null, version);
        this.scriptSQLCreate = scriptSQLCreate;
        this.scriptSQLDelete = scriptSQLDelete;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        int qtdScript = scriptSQLCreate.length;
        for (int i = 0; i < qtdScript; i++) {
            String sql = scriptSQLCreate[i];
            Log.i("SQL_INIT", sql);
            db.execSQL(sql);
        }

    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int versaoAntiga, int versaoNova) {
        db.execSQL(scriptSQLDelete);
        onCreate(db);

    }

    public void onDelete(SQLiteDatabase db) {
        int qtdScript = scriptSQLCreate.length;
        for (int i = 0; i < qtdScript; i++) {
            String sql = scriptSQLCreate[i];
            db.execSQL(sql);
        }
    }
}

File: app/src/main/java/br/com/onibusapp/onibusapp/data/SQLiteHelper.kt
package br.com.onibusapp.onibusapp.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log

/**
 * Created by diego on 18/07/2018.
 */

class SQLiteHelper(context: Context, name: String, version: Int, private val scriptSQLCreate: Array<String>,
                   private val scriptSQLDelete: String) : SQLiteOpenHelper(context, name, null, version) {

    override fun onCreate(db: SQLiteDatabase) {
        val qtdScript = scriptSQLCreate.size
        for (i in 0 until qtdScript) {
            val sql = scriptSQLCreate[i]
            Log.i("SQL_INIT", sql)
            db.execSQL(sql)
        }

    }

    override fun onUpgrade(db: SQLiteDatabase, versaoAntiga: Int, versaoNova: Int) {
        db.execSQL(scriptSQLDelete)
        onCreate(db)

    }

    fun onDelete(db: SQLiteDatabase) {
        val qtdScript = scriptSQLCreate.size
        for (i in 0 until qtdScript) {
            val sql = scriptSQLCreate[i]
            db.execSQL(sql)
        }
    }
}
